File: app/src/main/java/demo/victormunoz/githubusers/features/userDetails/UserDetailsContract.java
package demo.victormunoz.githubusers.features.userDetails;

import demo.victormunoz.githubusers.model.entity.User;

public interface UserDetailsContract {

    interface PresenterListener {

        void displayUserDetails(User user);

        void onLoadUserDetailsFail();
    }

    interface UserActionsListener {

        void getUserDetails(String login);

        void onBackPressed();
    }
}

File: app/src/main/java/demo/victormunoz/githubusers/features/userDetails/UserDetailsContract.kt
package demo.victormunoz.githubusers.features.userDetails

import android.graphics.Bitmap
import demo.victormunoz.githubusers.model.User

interface UserDetailsContract {

    interface ActivityListener {

        fun displayUserDetails(user: User, bitmap: Bitmap)

        fun showErrorMessage()
    }

    interface ActivityPresenterListener {

        fun getUserDetails(login: String, url: String, imageWidth: Int)

        fun onBackPressed()
    }
}
